lib/tail-file.js

```diff
@@ -209,7 +211,9 @@ class TailFile extends Readable {
         await this._streamFileChanges()
         if (this[kStream]) return // Pause polling if backpressure is on
       } else {
-        setImmediate(this.emit.bind(this), 'flush')
+        setImmediate(this.emit.bind(this), 'flush', {
```

Comment:
This flush is basically for quitting when there is no data to read, but I don't think I like this. It creates 2 flush events which can cause double callbacks. We might choose to fix this in another commit, but basically I should be able to:
1) .start()
2) await flush
3) .quit()

Instead, there's as loop between 2 & 3 unless the developer wisely uses `once(tail, 'flush')`

Let's think about this (and read what had to be done in the test to avoid this). Perhaps anew event only called `quit` that can be awaited in `.quit()`.

Reply:
I will open a new issue with a working example of this problem and we can address it as a `major` change

Reply:
Opened https://github.com/logdna/tail-file-node/issues/24